unitTests.js

```diff
@@ -0,0 +1,46 @@
+describe('NotesCollectionView', function(){
+  context('when a list item is added to a note', function(){
+    var collection = $el = view = null;
+
+    //setup the view
+    beforeEach(function(){
+      collection = new NotesCollection();
+      collection.createCallCount = 0;
+      collection.create = function(){
+        this.createCallCount++;
+      };
+      $el = $('<div />');
+      view = new NotesCollectionView({
+        collection: collection,
+        el: $el
+      });
+      view.render();
+    });
+
+    //add an input & click button
+    beforeEach(function(){
+      view.$('#notesInput').val('List Item');
+      view.$('#tskBtn').click();
+    })
+
+    //teardown the view
+    afterEach(function(){
+      $el.remove();
+    });
+
+    it('resets input field to nothing', function(){
+      chai.expect(view.$('#notesInput').val()).to.equal('');
```

Comment:
OK, I stopped being lazy and actually pulled this down to check.

The test actually looks ok. The issue's with the implementation [here](https://github.com/RouteRunner/RouteRunnerApp/blob/master/js/waypointsInput.js#L95), and it's interesting one. Notice that the test is checking `view.$('#notesInput')`, while the implementation is using a "naked" (aka global) `$`? Changing the implementation to scope jQuery to `this.$` will fix the test.

Here's why: jQuery elements by default are detached from the global DOM. That's the case when we create a root element via `$('<div />')`, and it provides some nice separation in the test. But there's a trap: *lookups* using the *global* jQuery operator (`$`) don't know what DOM fragment to use as a root node, and end up falling back on the global DOM. Since the root element we provided in the test setup *isn't* in the global DOM, this lookup is failing and finding some other element matching `#tskBtn`.

Note that we could also fix this test by attaching `view.$el` to the global DOM during test setup (`view.$el.appendTo('body')`) and remove it after we're done... but in this case the isolated test case has actually helped us catch a "leaky" global variable and prevent any future bugs it might cause.